src/status_im/contexts/keycard/events.cljs

```diff
@@ -0,0 +1,62 @@
+(ns status-im.contexts.keycard.events
+  (:require [re-frame.core :as rf]
+            status-im.contexts.keycard.login.events
+            status-im.contexts.keycard.pin.events
+            status-im.contexts.keycard.sheet.events
+            [taoensso.timbre :as log]))
+
+(rf/reg-event-fx :keycard.callback/check-nfc-enabled-success
+ (fn [{:keys [db]} [nfc-enabled?]]
+   {:db (assoc-in db [:keycard :nfc-enabled?] nfc-enabled?)}))
+
+(rf/reg-event-fx :keycard.ios.callback/on-nfc-user-cancelled
+ (fn [{:keys [db]}]
+   (log/debug "[keycard] nfc user cancelled")
+   {:db (-> db
+            (assoc-in [:keycard :pin :status] nil))
+    :fx [(when-let [on-nfc-cancelled-event-vector (get-in db [:keycard :on-nfc-cancelled-event-vector])]
+           [:dispatch on-nfc-cancelled-event-vector])]}))
+
+(rf/reg-event-fx :keycard.callback/on-card-connected
+ (fn [{:keys [db]} _]
+   (log/debug "[keycard] card globally connected")
+   {:db (assoc-in db [:keycard :card-connected?] true)
+    :fx [(when-let [event (get-in db [:keycard :on-card-connected-event-vector])]
+           [:dispatch event])]}))
+
+(rf/reg-event-fx :keycard.callback/on-card-disconnected
+ (fn [{:keys [db]} _]
+   (log/debug "[keycard] card disconnected")
+   {:db (assoc-in db [:keycard :card-connected?] false)
+    :fx [(when-let [event (get-in db [:keycard :on-card-disconnected-event-vector])]
+           [:dispatch event])]}))
+
+(rf/reg-event-fx :keycard.ios/start-nfc
+ (fn [_]
```

Comment:
why is bad to have them ?

Reply:
It's perfectly fine and it's actually a requirement in Clojure JVM, but it's uncommon in `status-mobile` to put underscores in event handler signatures, probably because CLJS allows this otherwise we would get compilation errors. Just a matter of code consistency.